### PR TITLE
Added extra styling, type notation, Tab component, and simple e2e tests

### PR DESCRIPTION
**Pull request by j-tran**
### Major
* Styling: Added various `.css` changes and font-awesome icons.
* Types: Added more notations and moved complicated object types to a separate `./app/types/` folder for composability.
* `Tab.js`: New component at`./app/components/Tab.js` which renders breadcrumbs based on the currently viewed table.
  * Had to add a `databaseName` property to a table object since haven't figured out a any other way to get add the database name to breadcrumb.
* Simple e2e tests: Created rudimentary tests in `e2e.spec.js` for opening and closing application.
### Minor
* Edited .flowconfig to ignore `./node_modules`

### CODE REVIEW

File: app/components/Tab.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (amilajack) [inline app/components/Tab.js]: Missing `@flow` pragma

File: app/components/Tab.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import type { TableType } from '../types/tableType';
+import styles from './Tab.css';
+import '../app.global.css';
+
+const Tab = ({ table }: Object<?TableType>) => {
```
Comment (amilajack) [inline app/components/Tab.js]: I don't think `Object<?TableType>) ` is a valid type definition. Are you trying to do `?TableType`?

Comment (amilajack) [inline app/components/Tab.js]: Two new lines after all imports

File: app/containers/Home.js [hunk 3]
```diff
@@ -3,18 +3,15 @@ import React, { Component } from 'react';
 import Table from '../containers/Table';
 import SideBar from '../containers/SideBar';
 import NavBar from '../containers/NavBar';
+import Tab from '../components/Tab';
+import type { TableType } from '../types/tableType';
+import type { DatabaseType } from '../types/databaseType';
 import styles from './Home.css';
 
 export default class HomePage extends Component {
```
Comment (amilajack) [inline app/containers/Home.js]: Two new lines after import

File: app/containers/SideBarItem.js [hunk 4]
```diff
@@ -1,30 +1,54 @@
+// @flow
 import React, { Component } from 'react';
 import styles from './SideBar.css';
+import type { TableType } from '../types/tableType';
+import '../app.global.css';
+
```
Comment (amilajack) [inline app/containers/SideBarItem.js]: 2 nl after import

File: app/containers/SideBar.js [hunk 5]
```diff
@@ -1,13 +1,16 @@
 // @flow
 import React, { Component } from 'react';
 import SideBarItem from './SideBarItem';
+import type { DatabaseType } from '../types/databaseType';
+import type { TableType } from '../types/tableType';
 import styles from './SideBar.css';
 
-
```
Comment (amilajack) [inline app/containers/SideBar.js]: 2 nl after import

File: app/containers/SideBarItem.js [hunk 6]
```diff
@@ -1,30 +1,54 @@
...
       </li>
     );
 
     return (
       <div>
-        <li onClick={this.onDatabaseClick} className={styles.sidebarItem}>
-          <a href="#">{this.props.name}</a>
+        <li className={styles.sidebarItem}>
+          <a onClick={this.onDatabaseClick} href="#">
+            {this.state.tablesVisible ?
```
Comment (amilajack) [inline app/containers/SideBarItem.js]: Ternary style should be like so:
```js
condition
  ? foo()
  : bar()
```

File: app/types/tableType.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// @flow
+export type TableType = {
+  // TODO: Temporary. Not sure how else to store the state
```
Comment (amilajack) [inline app/types/tableType.js]: Create special comments like so:
```js
// @TODO: Some foo bar baz that breaks down to next line bc so long
//        Foo bar super cool baz
```

File: .eslintrc [hunk 8]
```diff
@@ -35,8 +38,7 @@
   ],
   "settings": {
     "import/resolver": {
-      "webpack": {
-        "config": "webpack.config.eslint.js"
+      "webpack": {"config": "webpack.config.eslint.js"
```
Comment (amilajack) [inline .eslintrc]: nit: Previous style seems more readable

File: app/components/TableRow.js [hunk 9]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
```
Comment (amilajack) [inline app/components/TableRow.js]: Missing flow annotation. Try adding the `eslint-plugin-flowtype` rule that checks for missing flowtype annotations

File: app/containers/NavBar.js [hunk 10]
```diff
@@ -2,13 +2,10 @@
...
 import styles from './NavBar.css';
 
-type AppState = {
-  sections: Array<string>
-};
 
 export default class NavBar extends Component {
 
-  state: AppState;
+  state: {sections: Array<string>};
```
Comment (amilajack) [inline app/containers/NavBar.js]: nit: Convention is to break down object literal braces. Codebase gets hard to skim quickly if code styles are not consistent

Comment (j-tran) [inline app/containers/NavBar.js]: So the style should be like this?

```
state: {
    sections: Array<string>
};
```

Comment (amilajack) [inline app/containers/NavBar.js]: Yup 👍

**Comment by amilajack** [comment]
🎉 🎉 🎉 🎉 🎉

**Pull request merged**